gcsio/src/main/java/com/google/cloud/hadoop/gcsio/EventLoggingHttpRequestInitializer.java

```diff
@@ -0,0 +1,89 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.MapMaker;
+import com.google.common.flogger.GoogleLogger;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+class EventLoggingHttpRequestInitializer implements HttpRequestInitializer {
```

Comment:
I don't see this request initializer getting used anywhere ? Is that intentional ? 

I think we are missing the wiring part of `isTraceEnabled`

Reply:
Yes, it will be separate change. Split this into two that change will have conflict b/w master and 2.2.x branch and it would be cleaner to separate them into to PRs.